src/units/ability.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+	Copyright (C) 2003 - 2022
+	by David White <dave@whitevine.net>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "config.hpp"
+#include "units/ptr.hpp"
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+using ability_vector = std::vector<ability_ptr>;
+
+class unit_ability_t
```

Comment:
I'm not quite sure about this. Is the `_t` suffix consistent with our other classes? It feels inconsistent to me.